src/iop/channelmixerrgb.c

```diff
@@ -687,7 +687,7 @@ static inline void loop_switch(const float *const restrict in, float *const rest
       case DT_ADAPTATION_FULL_BRADFORD:
       {
         // Convert from RGB to XYZ
-        dot_product(temp_two, RGB_to_XYZ, temp_one);
+        dt_apply_transposed_color_matrix(temp_two, RGB_to_XYZ, temp_one);
```

Comment:
why ?

Reply:
I think the dot_product -> dt_apply_transposed_color_matrix stem from https://github.com/darktable-org/darktable/pull/10963 and that pr has all the explaination for it.

Reply:
Yep, as I had to deal with color matrices here and already had the optimizations of https://github.com/darktable-org/darktable/pull/10963 done but not merged yet, I included the commits here as well. These changes can be safely ignored.